activerecord/lib/active_record/associations/preloader/through_association.rb

```diff
@@ -82,7 +82,9 @@ def through_scope
               scope.where! reflection.foreign_type => options[:source_type]
             else
               unless reflection_scope.where_clause.empty?
-                scope.includes_values = Array(reflection_scope.values[:includes] || options[:source])
+                include_values = Array(reflection_scope.values[:includes] || options[:source])
+                scope.includes_values = include_values
+                scope.references! include_values
```

Comment:
This seems over-broad.. we're already pulling in the scope's `references` below, so I think this only needs `:source`?

TBH, I'm a bit confused about the logic behind the "includes || source" thing in the first place (wouldn't we need to include the source regardless of whether there are other includes defined?), so I may be missing something, or misunderstanding what these values look like.